spec/data-types-and-values.html

```diff
@@ -7,12 +7,35 @@ <h1>ECMAScript Language Types</h1>
 
     <emu-clause id="sec-ecmascript-language-types-record-type">
       <h1>The Record Type</h1>
-      <p>TBD</p>
+      <p>
+        The Record type is the set of all sorted sequences of zero or more pairs of key and data up to a maximum length of 2<sup>53</sup> - 1 elements. Pairs have the form { [[Key]], [[Value]] }, where [[Key]] is a String and [[Value]] is any primitive (including Record and Tuple). Pairs are sorted by [[Key]] in code unit order. Operations which create Record values are responsible for setting this up appropriately. The List of pairs is never modified.
+      </p>
+      <p>
+        The Record type is used to represent the association of multiple primitive values together. Each pair is regarded as occupying a position within the sequence. These positions are indexed with nonnegative integers. The first pair (if any) is at index 0, the next element (if any) at index 1, and so on. The length of a Record is the number of pairs within it. The empty Record has length zero and therefore contains no elements.
+      </p>
+
+      <emu-clause id="sec-recordtypecreate" aoid="RecordTypeCreate">
+        <h1>RecordTypeCreate ( _keysList_, _valuesList_ )</h1>
+        <p>This abstract operation RecordTypeCreate takes a _keysList_ and a _valuesList_ argument. It is used to specify the creation of new Record value types. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: _keysList_ is a List of Strings.
+            1. Assert: _valuesList_ is a List of primitive values.
+
+            1. Assert: _valuesList_.length == _keysList_.length
+            1. Let _L_ be an empty List.
+            1. For each _i_ from 0 to _keysList_.length:
+              1. Add { [[Key]]: _keysList_[_i_], [[Value]]: _valuesList_[_i_] } to _L_
+            1. Sort _L_ by [[Key]] using code unit order
+            1. Return _L_
+        </emu-alg>
+      </emu-clause>
     </emu-clause>
 
     <emu-clause id="sec-ecmascript-language-types-tuple-type">
       <h1>The Tuple Type</h1>
-      <p>TBD</p>
+      <p>

```

Comment:
At some point, you'll want to look at [Integer Indexed exotic objects](https://tc39.es/ecma262/#sec-integer-indexed-exotic-objects) to inform how property access works here.